src/main/groovy/net/vivin/gradle/versioning/SemanticBuildVersion.groovy

```diff
@@ -19,7 +19,7 @@ class SemanticBuildVersion {
 
     String snapshotSuffix = "SNAPSHOT"
 
-    Pattern tagPattern = ~/.*/
+    Pattern tagPattern = ~/\d++\.\d++\.\d++/
```

Comment:
Well, you don't need the `^.*` part, because you use `find()` and not `matches`. It would just decrease the performance of the RE, as there is more to backtrack and remember. 

The second `+` is also a performance tweak that you can use if you know what you are doing. It can be added to any quantifier and makes it possessive. This means it does not give up matched characters when backtracking and so the matching fails faster if no match can be found if used correctly. 

One example: If you find `^[ab]++b` on the string `abb` it will **not** match. If you try with `^[ab]+b` you will match `abb`. If you try with `^[ab]+?b`, you will match `ab`. In the first variant, first `[ab]++` will match `abb`, now there is nothing left `b` could match on and as the quantifier is possessive it does not give up matched characters and thus matching ends without match. In the second variant, the `[ab]++` gives up one character after `b` did not match and `b` can match. In the third variant the quantifier is reluctant and thus matches as few as possible, eating more characters during backtracking if necessary and thus `ab` is matched in the first run immediately. 

Now if you have an RE like the one I used, you can easily use the possessive quantifier to improve performance, because `.` cannot appear in something that matches against `\d+`, so it does not make sense to give up characters during backtracking, as they will never match `\.`, so you can tell the RE engine to not even try which saves keeping the backtracking information and also the time to try many useless variants.